ddl/partition.go

```diff
@@ -634,16 +635,49 @@ func checkPartitionKeysConstraint(sctx sessionctx.Context, partExpr string, idxC
 	return nil
 }
 
-func extractPartitionColumns(sctx sessionctx.Context, partExpr string, tblInfo *model.TableInfo) ([]*expression.Column, error) {
-	e, err := expression.ParseSimpleExprWithTableInfo(sctx, partExpr, tblInfo)
+type columnNameExtractor struct {
```

Comment:
Please add some unit tests for this new structure.

Reply:
I think that the current tests covered it? It's a code change, not some new codes